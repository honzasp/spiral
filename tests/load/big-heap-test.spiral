(program
  (import std)
  (import std.test)
  (import std.heap)

  (fun list-of-ints (n)
    (do ((i n (- i 1))
         (xs null (cons i xs)))
        ((< i 0) xs)))

  (test "big heap of integers from list" (lambda (t)
    (let ((prime 104729)
          (xs (do ((i 0 (+ i 1))
                   (xs null (cons (mod (* i 123) prime) xs)))
                  ((>= i prime) xs)))
          (h (heap-from-list - xs)))
      (assert-equal t (heap-to-list h) (list-of-ints (- prime 1))))))

  (test "big heap of integers to list" (lambda (t)
    (let ((prime 104729)
          (h (do ((i 0 (+ i 1))
                  (h (heap-new -)
                     (heap-insert h (mod (* i 123) prime))))
                 ((>= i prime) h))))
      (assert-equal t (heap-to-list h) (list-of-ints (- prime 1)))))))

