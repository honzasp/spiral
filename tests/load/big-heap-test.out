test big heap of integers from list ok
test big heap of integers to list ok
